### ISSUE TITLE
Make the crate play nicely with `colstodian`

### ISSUE THREAD

**Issue by virtualritz** [virtualritz/after-effects#48]
See subject. Or, alternatively, with [`colstodian`](https://docs.rs/colstodian/latest/colstodian/)'s base, [`kolor`](https://docs.rs/kolor/latest/kolor/).

Color management in Ae is still a mess. At least on the SDK side we can do better in Rust.

Whichever is the better suited abstraction.

**Comment by virtualritz** [comment]
## Recommendation: color-management interop

Did a research pass across the candidate crates and the actual AE pixel types. Summary and a concrete proposal below.

### TL;DR

1. **Do it -- but lead with Linebender's `color` crate, not `colstodian`.** `color` is the only candidate that is actively maintained, publishable against, and whose model fits AE: `AlphaColor<CS>` (straight) vs `PremulColor<CS>` (premultiplied) as distinct types, `AcesCg`/`Aces2065_1` built in alongside sRGB/LinearSrgb/DisplayP3/Rec2020/Oklab, and `DynamicColor` (a runtime color-space tag -- the only construct in any candidate that matches AE's runtime-determined working space).
2. **`colstodian`/`kolor` interop is currently impossible on crates.io.** The active code lives in the `virtualritz/colstodian` fork, which git-depends on the `virtualritz/kolor` fork; a crates.io-published `after-effects` cannot carry a git dependency. The crates.io names are held by others (`colstodian` = fu5ha, last real publish 2021; `kolor` = kabergstrom, 2023), and both upstreams are effectively dormant. So `colstodian` interop is **blocked on publishing the forks** (upstream `kolor` has merged your PRs -- an ownership transfer may be the shortest path). The feature design below makes adding it later non-breaking.
3. **Orphan-rule surprise:** the `nalgebra` precedent (`impl From<Matrix4> for nalgebra::Matrix4<f64>`) does **not** transfer. `Matrix4` is a local wrapper, but the pixel types are deliberately raw `ae_sys` re-exports (`pf/pixel.rs` avoids wrappers for perf), so `From` between two foreign types is orphan-blocked. Interop must be a small **extension trait**.

### What AE pixels actually are

| Alias | C type | Layout | Range |
|---|---|---|---|
| `Pixel8` | `PF_Pixel` | `{ alpha, red, green, blue }: u8` (ARGB order) | 0..=255 |
| `Pixel16` | `PF_Pixel16` | same, `u16` | **0..=32768** (`PF_MAX_CHAN16`, not 65535) |
| `PixelF32` | `PF_Pixel32` | same, `f32` | ~0..1, HDR over-range allowed |
| `PixelF64` | `AEGP_ColorVal` | `{ alphaF.. }: f64` | 0..1 |

Semantics interop must respect: alpha is **straight/unmatted by default** (AE composites straight; the premultiply utilities exist because buffers normally are not premultiplied); and the **color space is a runtime property of the project** (working space / OCIO -- already wrapped in `aegp::suites::ColorSettings`), so no compile-time `From` can honestly pick a space. The caller must assert it.

### Crate ranking

1. **`color` (Linebender) -- recommend now.** 0.3.x, actively developed, zero mandatory deps, `no_std`-capable. Straight/premul as distinct types, ACES built in, `components: [f32;4]` maps to `PixelF32` directly, `DynamicColor` matches AE's runtime working space, `Rgba8` covers the 8-bit path.
2. **`colstodian` (your fork) -- right idea, defer.** Most correctness-enforcing (`Color<E: ColorEncoding>`), and it's your fork. Blockers: git dep on the kolor fork (unpublishable); crates.io name divergence; and **alpha only exists in the sRGB family** -- `AcesCg`/`Aces2065`/`Bt2020`/`DisplayP3` encodings are alpha-less `Vec3`, so an AE float ARGB pixel in an ACEScg working space cannot round-trip alpha. Revisit once forks are published and gain alpha-bearing working-space encodings.
3. **`cint` -- optional neutral bridge.** Zero-dep "mint of color" with `Alpha`/`PremultipliedAlpha` wrappers; future-proofs interop with anything implementing `ColorInterop`. But the colstodian 0.2 fork does not currently implement `cint` (dep present, unused), so transitive value is speculative today.
4. **`palette` -- only on demand.** Mature but display-referred, no ACES, heavy generics, and assumes full-range u16 (mismatches AE's 0..32768).
- **Not recommended directly: `kolor`** -- it's conversion math over alpha-less `Vec3`, a backend not a pixel surface; if it arrives it should arrive transitively via colstodian.

### Feature-flag design -- additive, never mutually exclusive

Independent `impl` blocks, exactly like the existing (implicit) `nalgebra`/`ultraviolet` interop. Name each feature after its crate:

```toml
# after-effects/Cargo.toml
color = { version = "0.3", optional = true, default-features = false, features = ["std"] }
# later, when publishable:
# colstodian = { version = "0.2", optional = true }
# cint       = { version = "0.3", optional = true }
```

### Sketch (extension trait, orphan-rule-safe)

```rust
#[cfg(feature = "color")]
mod color_interop {
    use super::*;
    use color::{AlphaColor, ColorSpace, PremulColor, Rgba8};

    /// Lossless layout/range conversion between AE pixels and `color` types.
    /// These only reorder ARGB->RGBA and rescale components -- NO color-space
    /// transform. The caller asserts the space `CS`; AE pixel data is in the
    /// project working space (a runtime property, see aegp::ColorSettings), and
    /// AE buffers carry straight alpha unless you premultiplied them.
    pub trait PixelInterop: Sized {
        fn to_alpha_color<CS: ColorSpace>(self) -> AlphaColor<CS>;
        fn from_alpha_color<CS: ColorSpace>(c: AlphaColor<CS>) -> Self;
        fn to_premul_color<CS: ColorSpace>(self) -> PremulColor<CS>;
        fn from_premul_color<CS: ColorSpace>(c: PremulColor<CS>) -> Self;
    }

    impl PixelInterop for PixelF32 {
        #[inline]
        fn to_alpha_color<CS: ColorSpace>(self) -> AlphaColor<CS> {
            AlphaColor::new([self.red, self.green, self.blue, self.alpha])
        }
        #[inline]
        fn from_alpha_color<CS: ColorSpace>(c: AlphaColor<CS>) -> Self {
            let [red, green, blue, alpha] = c.components;
            PixelF32 { alpha, red, green, blue }
        }
        // premul variants identical modulo type
    }
    // Pixel16: scale by 1.0/32768.0 (clamp on the way back); Pixel8: 1.0/255.0;
    // PixelF64: f64<->f32. Plus an 8-bit fast path: Pixel8 <-> color::Rgba8 (pure swizzle).
}
```

Usage reads honestly:

```rust
use color::{AcesCg, Srgb};
let c   = in_pixel.to_alpha_color::<AcesCg>();  // "my working space is ACEScg"
let out = c.convert::<Srgb>();                  // color does the math
*out_pixel = PixelF32::from_alpha_color(out);
```

Optional nice-to-have no other crate can express: a helper on `aegp::ColorSettings` that reads the runtime working space / OCIO name and returns a `color::DynamicColor` tag.

### Risks / decisions needed

1. **The crux:** until `virtualritz/colstodian` + `virtualritz/kolor` are on crates.io, `after-effects` cannot depend on them. Decision: pursue publishing, or ship `color` interop and revisit.
2. **No color-space guessing** -- keep `CS` caller-asserted; resist an `assume_srgb()` default (breaks the moment a project is color-managed/linearized).
3. **Premultiplication:** conversions relabel, never (un)multiply -- point users at `color`'s `premultiply()`/`un_premultiply()` and the existing `UtilCallbacks::premultiply`/`FillMatteSuite`.
4. **16-bit range:** hard-code the 0..32768 scale (reuse `PF_MAX_CHAN16` like `pixel8_to_16`) + a round-trip test.
5. **Version churn:** `color` is pre-1.0 (0.3.x) -- a bump is a semver-visible public-API change, same trade-off already accepted for `nalgebra`/`ultraviolet`.
6. Premiere packed/YUV formats (`pr::PixelFormat` VUYA etc.) out of scope for v1.
7. `PixelF64` through an f32 crate loses precision -- fine (AE renders f32), worth a doc note.

**Bottom line:** implement the `color` feature now (small, additive, matches the `nalgebra` precedent modulo the orphan-rule extension trait); note that `colstodian` interop is blocked on crates.io publication of the forks and on the fork gaining alpha-bearing working-space encodings; leave `cint`/`palette`/`colstodian` as future additive features under the same pattern.
